simulation/models/searchQueries.js

```diff
@@ -0,0 +1,38 @@
+const mongoose = require('../db');
+
+const searchQuerySchema = mongoose.Schema({
+  searchQueryId: Number,
+  timestamp: Date,
+  visitId: Number,
+  userId: Number,
+  market: String,
+  checkIn: Date,
+  checkOut: Date,
+  roomType: String,
+});
+
+searchQuerySchema.index({ searchQueryId: 1 }, { unique: true });
+
+const SearchQuery = mongoose.model('SearchQuery', searchQuerySchema);
+
+const emptySearchQueries = () =>
+  new Promise((resolve, reject) => {
```

Comment:
Mongoose supports native promises. See http://mongoosejs.com/docs/promises.html

Reply:
I'll test this out and if it works, I'll will update this in my next PR.